ports/models.py

```diff
@@ -28,6 +33,236 @@ class Port(models.Model):
 
     objects = PortManager()
 
+    @classmethod
+    def load(cls, data):
+        def open_portindex_json(path):
+            with open(path, "r") as file:
```

Comment:
Let's just catch the error in case the `path` does not exist and notify the user instead of iterating through all the functions and doing nothing.

Reply:
`path` is passed on to this function by a management command- which throws an error if the file was not found.

If we need to catch the error here also, I will add it.

Reply:
Done.

Reply:
Now that I think about your comment, you are right that it will simply fail at `open` and it seems like catching the exception here could be an over-the-top.

Reply:
Just use your best judgment here.

Reply:
I have added a check while calling the function `open_portindex_json`.